code/modules/spells/spell_types/shapeshift.dm

```diff
@@ -45,6 +45,26 @@
 			shapeshift_type = animal_list[shapeshift_type]
 
 		var/obj/shapeshift_holder/S = locate() in M
+		if(M.movement_type & (VENTCRAWLING))
+			var/mob/living/ventcrawlmaybe = shapeshift_type
+			if((S && !S.stored.ventcrawler) || !shapeshift_type.ventcrawler) //you're shapeshifting into something that can't fit into a vent
+				var/turf/turfyoudieon = get_turf(M)
+				var/obj/machinery/atmospherics/pipe/pipeyoudiein = locate() in turfyoudieon
+				if(!turfyoudieon || !pipeyoudiein) //not sure how this happens but sanity
+					return
+				to_chat(M, "<span class='userdanger'>Shapeshifting inside of [pipeyoudiein] quickly turns you into a bloody mush.</span>")
+				var/list/viable_vents = list() //list of vents and scrubbers in the pipeline
+				for(var/obj/machinery/atmospherics/components/unary/possiblevent in pipeyoudiein.parent.other_atmosmch)
+					if(istype(possiblevent, /obj/machinery/atmospherics/components/unary/vent_pump) || istype(possiblevent, /obj/machinery/atmospherics/components/unary/vent_scrubber))
+						viable_vents += possiblevent
+				for(var/i in 1 to 3)
```

Comment:
Could check for vents and scrubbers in a set range, then check if their pipenet matches your current pipenet. Since distro and waste is enormous, 3 random vents are likely nowhere near wherever this guy fucked up.

Something like
```
for(var/obj/machinery/atmospherics/components/unary/possiblevent in range(10, user))
    if(pipeyoudiein.parent == possiblevent.parent)
        viable_vents += possiblevent
```

Reply:
I'll revisit the actual mechanics here

Reply:
is this wip or what

Reply:
tbh who cares if it's realistic, it's funny.